extra/import_data.py

```diff
@@ -1,197 +1,158 @@
+            continue
+        else:
+            # retrieve basic info from Alissa about the analysis
+            patient_dn_no = client.get_report(analysis['id'])[0]['reportName'].split("_")[0]
+            logger.info('start Alissa retrieval of: %s' % patient_dn_no)
+            ONB01_analysis = client.get_inheritance_analyses(analysis['id'])
+            accession_number = client.get_patient(analysis['patientId'])['accessionNumber']
+            export_id = client.post_inheritance_analyses_variants_export(analysis['id'], marked_review=True, marked_include_report=False)['exportId']
+            analyis_sources = client.get_sources(analysis['id'])
+            
+            # retrieve the VUS/GUS info based on the analysis ID
+            VUS_response = None
+            while VUS_response is None:
+                try:
+                    time.sleep(5)  # 5 sec delay to request exported report.
+                    VUS_response = client.get_inheritance_analyses_variants_export(analysis['id'], export_id)
+                    #response.raise_for_status()
+                except HTTPError as exception:
+                    print(exception)
+                    pass
+            logger.info('Alissa retrieval completed of: %s' % patient_dn_no)
+            # sometimes the there are no VUS marked or found within an analysis, then no info needs to be uploaded
+            if VUS_response == []:
+                logger.info('Patient %s, has no VUS marked/found. Not uploaded to MongoDB' % patient_dn_no)
+                continue
+            else:
+                upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response)
+    # if latest date does not retrieve any analyses from Alissa, quit program and try later
+    if analysis is None:
+        logger.info('no new analysis in Alissa since last upload to VUSualizer')
+        sys.exit(0)
+    # after uploading all analyses to MongoDB, replace time of last upload with starttime of script (before calling Alissa)
+    db.replace_one({"version" : 1}, {"lastUpdatedOn" : start_time, "version" : 1}, True)
+
+
+def upload_to_mongodb(ONB01_analysis, accession_number, analyis_sources, patient_dn_no, VUS_response):
+    '''Function, for parsing data to the MongoDB'''
+    # connection with MongoDB and the correct database "vus" and collection "variant"
+    mongoclient = pymongo.MongoClient("mongodb://localhost:27017/")
+    db = mongoclient.vus.variant
+    
+    # if DNnr already in mongoDB, check if updated an replace if neccesary
+    if db.count_documents({"dn_no": patient_dn_no}) > 0:
+        logger.info('dn_no: %s already present' % patient_dn_no)
+        # get and parse "lastUpdatedOn" from MongoDB/Alissa for specific patient_dn_no
+        lastUpdatedOn_mongoDB = db.find_one({"dn_no" : patient_dn_no}, {"lastUpdatedOn":1, "_id":0})
+        for key, value in lastUpdatedOn_mongoDB.items():
+            lastUpdatedOn_mongoDB = value
```

Comment:
Would it be faster (and save you a query) if you immediately perform the find_one instead of first counting and then perform a find_one?

Reply:
I think this is the same as an earlier comment by yourself. I always use this way to read python dictionaries and do not know of a different way. I will have a look (because in essence I agree with what you are telling)

Reply:
I now see what you meant (my last comment did not address your point). But I tried your suggestion and it works just as well. so I changed it 👍 (was a remnant of the old Excel upload method)